api/kyverno/v2/updaterequest_types.go

```diff
@@ -82,23 +82,45 @@ type UpdateRequestSpec struct {
 	// Specifies the name of the policy.
 	Policy string `json:"policy" yaml:"policy"`
 
+	// RuleContext is the associate context to apply rules.
+	// optional
+	RuleContext []RuleContext `json:"ruleContext,omitempty" yaml:"ruleContext,omitempty"`
+
 	// Rule is the associate rule name of the current UR.
 	Rule string `json:"rule" yaml:"rule"`
```

Comment:
It is used for mutate existing rules.

Reply:
Can we set the rule name in the rule context for mutateExisting instead of using a separate field dedicated to mutate existing rule name?

Another question, a UR is created per policy, what if we have two mutateExisting rules, which rule name would be associated with the UR?

Reply:
I should make it clear this change is for the generate policy only. I'm thinking of updating the same for the mutate existing policy because each rule takes a list of targets to be mutated. I'd like to keep this change for Generation only and I can send a follow-up PR to update mutate existing.

Reply:
Got it.